src/sentry/replays/tasks.py

```diff
@@ -168,3 +169,91 @@ def _delete_if_exists(filename: str) -> None:
         storage_kv.delete(filename)
     except NotFound:
         pass
+
+
+@instrumented_task(
+    name="sentry.replays.tasks.run_bulk_replay_delete_job",
+    queue="replays.delete_replay",
+    default_retry_delay=5,
+    max_retries=5,
+    silo_mode=SiloMode.REGION,
+    taskworker_config=TaskworkerConfig(namespace=replays_tasks, retry=Retry(times=5)),
+)
+def run_bulk_replay_delete_job(replay_delete_job_id: int, offset: int, limit: int = 100) -> None:
+    """Replay bulk deletion task.
+
+    We specify retry behavior in the task definition. However, if the task fails more than 5 times
+    the process will stop and the task has permanently failed. We checkpoint our offset position
+    in the model. Restarting the task will use the offset passed by the caller. If you want to
+    restart the task from the previous checkpoint you must pass the checkpoint explicitly.
+    """
+    job = ReplayDeletionJobModel.objects.get(id=replay_delete_job_id)
+
+    # If this is the first run of the task we set the model to in-progress.
+    if offset == 0:
+        job.status = "in-progress"
+        job.save()
+
+    # Delete the replays within a limited range. If more replays exist an incremented offset value
+    # is returned.
+    results = fetch_rows_matching_pattern(
+        project_id=job.project_id,
+        start=job.range_start,
+        end=job.range_end,
+        query=job.query,
+        environment=job.environments,
+        limit=limit,
+        offset=offset,
+    )
+
+    # Delete the matched rows if any rows were returned.
+    if len(results["rows"]) > 0:
+        delete_matched_rows(job.project_id, results["rows"])
+
+    # Compute the next offset to start from. If no further processing is required then this serves
+    # as a count of replays deleted.
+    next_offset = offset + len(results["rows"])
+
+    if results["has_more"]:
+        # Checkpoint before continuing.
+        job.offset = next_offset
+        job.save()
+
+        run_bulk_replay_delete_job.delay(job.id, next_offset)
+        return None
+    else:
+        # If we've finished deleting all the replays for the selection. We can move the status to
+        # completed and exit the call chain.
+        job.offset = next_offset
+        job.status = "completed"
+        job.save()
+        return None
+
+
+@instrumented_task(
+    name="sentry.replays.tasks.delete_replay",
+    queue="replays.delete_replay",
+    default_retry_delay=5,
+    max_retries=5,
+    silo_mode=SiloMode.REGION,
+    taskworker_config=TaskworkerConfig(namespace=replays_tasks, retry=Retry(times=5)),
```

Comment:
```suggestion
    taskworker_config=TaskworkerConfig(
        namespace=replays_tasks,
        retry=Retry(times=5),
        processing_deadline_duration=120
    ),
```

This looks like it could take more than 10s to run.